webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/CampaignApi.scala

```diff
@@ -57,8 +58,11 @@ class CampaignApi(
     val schema = API.GetCampaigns
 
     def process0(version: ApiVersion, path: ApiPath, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
-      val res = (for {
-        campaigns  <- campaignRepository.getAll()
+      val campaignType  = req.params.get("campaignType").getOrElse(Nil).map(campaignSerializer.campaignType)
+      val (err, status) = req.params.get("status").getOrElse(Nil).map(v => CampaignStatus.getValue(v)).partitionMap(identity)
```

Comment:
too much go :)
Still, you need to deal with `err` (because we are not really go :) )